File: gm-utils/js/src/commom/utils/object/Obrig.js
import Null from './Null';

export default class Obrig {

	static check(o) {
		return Obrig.checkWithMessage(o, "Obrig.check: o is null");
	}

	static checkWithMessage(o, message) {
		if (Null.is(o)) {
			throw new Error(message);
		}
		return o;
	}

	static checkWithDynamicMessage(o, getMessage) {
		if (Null.is(o)) {
			throw new Error(getMessage());
		}
		return o;
	}

}

File: gm-utils/src/main/java/src/commom/utils/object/Obrig.java
package src.commom.utils.object;

import gm.utils.lambda.F0;

public class Obrig {

	public static <T> T check(T o) {
		return checkWithMessage(o, "Obrig.check: o is null");
	}

	public static <T> T checkWithMessage(T o, String message) {
		if (Null.is(o)) {
			throw new RuntimeException(message);
		}
		return o;
	}

	public static <T> T checkWithDynamicMessage(T o, F0<String> getMessage) {
		if (Null.is(o)) {
			throw new RuntimeException(getMessage.call());
		}
		return o;
	}

}
